### PR TITLE
Add error handlers to code generation

### PR DESCRIPTION
**Pull request by chengluyu**
This PR adds error handlers to code generation. So if code generation crashes, it can still displayed inferred types.

### CODE REVIEW

File: js/src/main/scala/Main.scala [hunk 1]
```diff
@@ -138,13 +104,89 @@ object Main {
     )
   }
 
+  // Extract a separated function to do this.
```
Comment (LPTK) [inline js/src/main/scala/Main.scala]:
```suggestion
```

Comment (LPTK) [reply js/src/main/scala/Main.scala]: This comment line is pretty useless 😄

File: js/src/main/scala/Main.scala [hunk 2]
```diff
@@ -138,13 +104,89 @@ object Main {
...
+  // Returns `Left[Str]` if successful, `Right[Str]` if not.
```
Comment (LPTK) [inline js/src/main/scala/Main.scala]: Note that the convention is `Right` for successful (it also sounds better, being able to say "the _right_ result").

File: js/src/main/scala/Main.scala [hunk 3]
```diff
@@ -289,14 +331,24 @@ object Main {
           ctx += nme -> ty_sch
           declared.get(nme).foreach { sign =>
             // ctx += nme -> sign  // override with less precise declared type?
-            subsume(ty_sch, sign)(ctx, raise, TypeProvenance(d.toLoc, "def definition"))
+            subsume(ty_sch, sign)(
```
Comment (LPTK) [inline js/src/main/scala/Main.scala]: Same remark as in the other PR.

File: js/src/main/scala/Main.scala [hunk 4]
```diff
@@ -289,14 +331,24 @@ object Main {
...
+              TypeProvenance(d.toLoc, "def definition")
+            )
           }
           res ++= formatBinding(d.nme, ty_sch)
           results append S(d.nme) -> (getType(ty_sch).show)
         case d @ Def(isrec, nme, R(PolyType(tps, rhs))) =>
           val errProv = TypeProvenance(rhs.toLoc, "def signature")
-          val ty_sch = PolymorphicType(0, typeType(rhs)(ctx.nextLevel, raise,
-            tps.map(tp => tp.name -> freshVar(noProv/*FIXME*/)(1)).toMap))
+          val ty_sch = PolymorphicType(
```
Comment (LPTK) [inline js/src/main/scala/Main.scala]: Same remark as in the other PR.

**Review by LPTK — APPROVED** [review]
Ok, LGTM. Please go ahead and merge after making the minor changes. Thanks!

File: js/src/main/scala/Main.scala (hunk 1 above)
Comment (chengluyu) [reply js/src/main/scala/Main.scala]: Removed. 😄

File: js/src/main/scala/Main.scala (hunk 3 above)
Comment (chengluyu) [reply js/src/main/scala/Main.scala]: Changed.

File: js/src/main/scala/Main.scala (hunk 2 above)
Comment (chengluyu) [reply js/src/main/scala/Main.scala]: Changed.

File: js/src/main/scala/Main.scala (hunk 4 above)
Comment (chengluyu) [reply js/src/main/scala/Main.scala]: Reverted.

**Pull request merged**